flake.nix

```
@@ -255,23 +281,37 @@
               ln -s ${image} $image
               echo "file binary-dist $image" >> $out/nix-support/hydra-build-products
             '';
-      } // builtins.listToAttrs (map
-          (crossSystem: {
-            name = "nix-${crossSystem}";
-            value = nixpkgsFor.${system}.cross.${crossSystem}.nix;
-          })
+      } // builtins.listToAttrs (forAllPackagesList
+          (pkgName: {
+            name = "${pkgName}-static";
+            value = nixpkgsFor.${system}.static.${pkgName};
+          }))
+        // builtins.listToAttrs (lib.concatMap
+          (crossSystem: forAllPackagesList
+            (pkgName: {
+              name = "${pkgName}-${crossSystem}";
+              value = nixpkgsFor.${system}.cross.${crossSystem}.${pkgName};
+            }))
           crossSystems)
```

Let's simplify this to one Cartesian product on attrsets:

 - delete `forAllPackagesList`
 - GitHub won't let me make a suggestion that includes the 7 lines that come after this, but those can be deleted too.
 - this suggestion:
 
```suggestion
      } // lib.concatMapAttrs
        (pkgName: { }: {
          "${pkgName}-static" = nixpkgsFor.${system}.static.${pkgName};
          "${pkgName}-${crossSystem}" = nixpkgsFor.${system}.cross.${crossSystem}.${pkgName};
          "${pkgName}-${stdenvName}" = nixpkgsFor.${system}.stdenvs."${stdenvName}Packages".${pkgName};
        })
        { "nix" = { };
          "nix-store" = { };
          "nix-util" = { };
        })
```

---

Lists are no good, especially when all you need are attributes.